src/live.rs

```diff
@@ -187,440 +160,450 @@ pub fn pxe_ignition_unwrap(config: &PxeIgnitionUnwrapConfig) -> Result<()> {
+    regions: Vec<Region>,
+    #[serde(skip_serializing)]
+    args: String,
 }
 
-/// Wrapper for a file handle to the content being modified (for example, an
-/// ISO image).  Usually this is where we write our modifications, but if
-/// we're streaming to stdout, it's where we read the content from.  In the
-/// case of an output file, it can be modified in place or copied from
-/// another file.  If complete() is not called and the file was copied, the
-/// copy will be deleted on drop.
-impl ContentFile {
-    fn new(input_path: &str, output_path: Option<&String>) -> Result<Self> {
-        match output_path.map(|v| v.as_str()) {
-            None => Ok(Self::InPlace(
-                OpenOptions::new()
-                    .read(true)
-                    .write(true)
-                    .open(&input_path)
-                    .with_context(|| format!("opening {}", &input_path))?,
-            )),
-            Some("-") => {
-                // check this here as a convenience to the caller
-                if isatty(stdout().as_raw_fd()).context("checking if stdout is a TTY")? {
-                    bail!("Refusing to write binary data to terminal");
-                }
-                // read-only for safety
-                Ok(Self::ForStdout(
-                    OpenOptions::new()
-                        .read(true)
-                        .open(&input_path)
-                        .with_context(|| format!("opening {}", &input_path))?,
-                ))
-            }
-            Some(unwrapped_output_path) => {
-                let output_dir = Path::new(unwrapped_output_path)
-                    .parent()
-                    .with_context(|| format!("no parent directory of {}", unwrapped_output_path))?;
-                let input = OpenOptions::new()
-                    .read(true)
-                    .open(&input_path)
-                    .with_context(|| format!("opening {}", &input_path))?;
-                let mut output = tempfile::Builder::new()
-                    .prefix(".coreos-installer-temp-")
-                    .tempfile_in(output_dir)
-                    .context("creating temporary file")?;
-                input
-                    .copy_to(output.as_file_mut())
-                    .with_context(|| format!("copying {} to temporary file", input_path))?;
-                Ok(Self::Copied((
-                    output,
-                    Path::new(unwrapped_output_path).to_path_buf(),
-                )))
-            }
-        }
-    }
-
-    fn is_stdout(&self) -> bool {
-        matches!(self, Self::ForStdout(_))
+impl KargEmbedAreas {
+    pub fn exists_in(file: &mut File) -> Result<bool> {
+        let region = Region::read(
+            file,
+            32768 - COREOS_IGNITION_HEADER_SIZE - COREOS_KARG_EMBED_AREA_HEADER_SIZE,
+            8,
+        )
+        .context("reading karg embed magic number")?;
+        Ok(region.contents == COREOS_KARG_EMBED_AREA_HEADER_MAGIC)
     }
```

Comment:
Would it be cleaner to fold this into `for_file` and have that return a `Result<Option<Self>>` instead?